### PR TITLE
feat(amazon-cognito-identity-js): Adding Session param for DEVICE_SRP_AUTH and user agent fix

### PR DESCRIPTION
**Pull request by elorzafe**
#### Description of changes
- Adding `Session` param for `DEVICE_SRP_AUTH` challenge
- Fixing user agent version on `amazon-cognito-identity-js` package

#### Description of how you validated changes
- [x] Tested UMD bundle on a plain HTML App
- [x] Tested npm package with a modern Web Framework App 
- [x] Tested with React-Native App

#### Checklist

- [X] PR description included
- [ X `yarn test` passes
- [X] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [X] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/src/UserAgent.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
+import { getUserAgent } from "./Platform";
 // constructor
-function UserAgent() {}
+function UserAgent() { }
 // public
-UserAgent.prototype.userAgent = 'aws-amplify/0.1.x js';
+UserAgent.prototype.userAgent = getUserAgent();
```
Comment (hkjpotato) [inline packages/amazon-cognito-identity-js/src/UserAgent.js:6]: I am a bit confusing of the original comment about using `UserAgent` as "constructor", as I don' see it is instantiated with `new` anywhere. The original use of prototype chain is a bit weird as well, but any refactor is out of scope of this PR.

Just a small callout, [this line](https://github.com/aws-amplify/amplify-js/blob/main/packages/amazon-cognito-identity-js/enhance-rn.js#L5) also have a hard code `userAgent`, should we update it as well.

**Review by hkjpotato — COMMENTED** [review]
just minor callout

File: packages/amazon-cognito-identity-js/src/CognitoUser.js [hunk 2]
```diff
@@ -503,6 +503,7 @@ export default class CognitoUser {
 		}
 
 		if (challengeName === 'DEVICE_SRP_AUTH') {
+			this.Session = dataAuthenticate.Session;
```
[the remark this answers is not in the document]
Comment (hkjpotato) [reply packages/amazon-cognito-identity-js/src/CognitoUser.js:506]: we have confirmed that offline

File: packages/amazon-cognito-identity-js/src/UserAgent.js (hunk 1 above)
Comment (elorzafe) [reply packages/amazon-cognito-identity-js/src/UserAgent.js:6]: Good catch @hkjpotato ! I will fix that

Comment (elorzafe) [reply packages/amazon-cognito-identity-js/src/UserAgent.js:6]: I removed that code is not necessary there any more.

Thanks!

File: packages/amazon-cognito-identity-js/src/Platform/index.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+import { version } from './version';
+
+const BASE_USER_AGENT = `aws-amplify/${version}`;
+
+export const Platform = {
```
[the remark this answers is not in the document]
Comment (elorzafe) [reply packages/amazon-cognito-identity-js/src/Platform/index.js:17]: Just wanted to have consistency with what we do on core

**Review by hkjpotato — APPROVED** [review]
LGTM! thank you @elorzafe

**Pull request merged**